Marshal/MarshaledObject.swift

```diff
@@ -14,25 +14,33 @@
 import Foundation
 
 
-// MARK: - Types
-
-public typealias MarshaledObject = [String: AnyObject]
-
-
-// MARK: - Dictionary Extensions
+public protocol MarshaledObject {
+    subscript(key: KeyType) -> Any? { get }
+    func anyForKey(key: KeyType) throws -> Any
+    func valueForKey<A: ValueType>(key: KeyType) throws -> A
+    func valueForKey<A: ValueType>(key: KeyType) throws -> A?
+    func valueForKey<A: ValueType>(key: KeyType) throws -> [A]
+    func valueForKey<A: ValueType>(key: KeyType) throws -> [A]?
+    func valueForKey<A: ValueType>(key: KeyType) throws -> Set<A>
+    func valueForKey<A: ValueType>(key: KeyType) throws -> Set<A>?
+    func valueForKey<A: RawRepresentable where A.RawValue: ValueType>(key: KeyType) throws -> A
+    func valueForKey<A: RawRepresentable where A.RawValue: ValueType>(key: KeyType) throws -> A?
+    func valueForKey<A: RawRepresentable where A.RawValue: ValueType>(key: KeyType) throws -> [A]
+    func valueForKey<A: RawRepresentable where A.RawValue: ValueType>(key: KeyType) throws -> [A]?
+    func valueForKey<A: RawRepresentable where A.RawValue: ValueType>(key: KeyType) throws -> Set<A>
+    func valueForKey<A: RawRepresentable where A.RawValue: ValueType>(key: KeyType) throws -> Set<A>?
+}
 
```

Comment:
The only reason why we would need to possibly override `anyForKey` is so that someone could change the key separator. What if we just expose a key separator property instead of allowing the method to be overridden?